app/Http/Controllers/Administration/SettingsController.php

```diff
@@ -333,6 +334,26 @@ public function setCSS(SetCSSSettingRequest $request): void
 		}
 	}
 
+	/**
+	 * Takes the js input text and put it into `dist/custom.js`.
+	 * This allows admins to actually execute custom js code on their
+	 * Lychee-Laravel installation.
+	 *
+	 * @param SetJSSettingRequest $request
+	 *
+	 * @return void
+	 *
+	 * @throws InsufficientFilesystemPermissions
+	 */
+	public function setJS(SetJSSettingRequest $request): void
+	{
+		/** @var string $js */
+		$js = $request->getSettingValue();
+		if (Storage::disk('dist')->put('custom.js', $js) === false) {
+			throw new InsufficientFilesystemPermissions('Could not save JS');
```

Comment:
```suggestion
			if (Storage::disk('dist')->get('custom.js') !== $js) {
				throw new InsufficientFilesystemPermissions('Could not save JS');
			}
```
Sometimes the content is written but the code fails because of some bits which cannot be set on the file.
As a result we check that the content has been modified as per requested. If so we do not consider this a failure.